jpa/eclipselink.jpa.test.jse/src/it/java/org/eclipse/persistence/jpa/json/TestJson.java

```diff
@@ -0,0 +1,225 @@
+                    .setParameter("id", e.getId()).getSingleResult();
+            Assert.assertEquals(value, dbValue.getValue());
+        } finally {
+            if (em.getTransaction().isActive()) {
+                em.getTransaction().rollback();
+            }
+            em.close();
+        }
+    }
+
+    @Test
+    public void testCreateJsonArrayField() {
+        EntityManager em = emf.createEntityManager();
+
+        JsonArray value = Json.createArrayBuilder()
+                .add("JsonString")
+                .add(42)
+                .add(JsonValue.FALSE)
+                .build();
+
+        try {
+            em.getTransaction().begin();
+            JsonArrayEntity e = new JsonArrayEntity(1004, value);
+            em.persist(e);
+            em.flush();
+            em.getTransaction().commit();
+            em.clear();
+
+            JsonArrayEntity dbValue = em.createQuery(
+                            "SELECT v FROM JsonArrayEntity v WHERE v.id=:id", JsonArrayEntity.class)
+                    .setParameter("id", e.getId()).getSingleResult();
+            Assert.assertEquals(value, dbValue.getValue());
+        } catch (Throwable t) {
+            t.printStackTrace();
+            throw t;
+        } finally {
+            if (em.getTransaction().isActive()) {
+                em.getTransaction().rollback();
+            }
+            em.close();
+        }
+    }
+
+    private boolean usedToDatabaseColumn = false;
+    private boolean usedToEntityAttribute = false;
+
+    @Override
+    public void usedToDatabaseColumn() {
+        usedToDatabaseColumn = true;
+    }
+
+    @Override
+    public void usedToEntityAttribute() {
+        usedToEntityAttribute = true;
+    }
+
+    @Test
+    public void testAnnotationconverterUsage() {
+        EntityManager em = emf.createEntityManager();
+
+        JsonValue value = Json.createObjectBuilder()
+                .add("id", "1001")
+                .add("name", "Joe Wright")
+                .add("age", 49)
+                .build();
+
+        JsonTestConverter.setConverterStatus(this);
+
+        try {
+            em.getTransaction().begin();
+            JsonValueWithConverter e = new JsonValueWithConverter(1005, value);
+            em.persist(e);
+            em.flush();
+            em.getTransaction().commit();
+            em.clear();
+
+            JsonValueWithConverter dbValue = em.createQuery(
+                    "SELECT v FROM JsonValueWithConverter v WHERE v.id=:id", JsonValueWithConverter.class)
+                    .setParameter("id", e.getId()).getSingleResult();
+            Assert.assertEquals(value, dbValue.getValue());
+            Assert.assertTrue(usedToDatabaseColumn);
+            Assert.assertTrue(usedToEntityAttribute);
+        } finally {
+            if (em.getTransaction().isActive()) {
+                em.getTransaction().rollback();
+            }
+            em.close();
+        }
+    }
+
+}
```

Comment:
Is it possible to add there test like `SELECT * FROM table WHERE json_column = :json_parameter`?

Reply:
Added this test. Everything except MySQL native `JSON` type usage passed.